game_server_launcher/launcher.py

```diff
@@ -221,6 +221,9 @@ def handle_add_player_message(self, msg):
         if msg.ip:
             self.logger.info('launcher: login server added player %d with ip %s' % (msg.unique_id, msg.ip))
             modify_firewall('whitelist', 'add', msg.unique_id, msg.ip)
+            if self.game_controller:
+                self.game_controller.send(
+                    Launcher2GamePlayerInfo(msg.unique_id, msg.rank_xp, msg.eligible_for_first_win))
```

Comment:
I think this should happen also for local players, so it should not be inside the if.

Reply:
Moving it.